src/averageAges.js

```diff
@@ -1,61 +1,51 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let manArray = [];
+
+  century
+    ? manArray = people.filter((man) => man.sex === 'm'
+    && Math.ceil(man.died / 100) === century)
+
+    : manArray = people.filter((man) => man.sex === 'm');
+
+  return manArray.reduce(
+    (averageAge, person) => averageAge + (person.died - person.born),
+    0,
+  ) / manArray.length;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womenArray = [];
+  const motherNames = people.filter((woman) => woman.mother !== null)
+    .map((a) => a.mother);
```

Comment:
```suggestion
  const motherNames = people
    .filter((woman) => woman.mother !== null)
    .map((a) => a.mother);
```

Reply:
but if you have huge array of people, your code will be slow

you create two loops here and after check with includes if it's mother. You can instead of include on 26 line use some method to check has someone mother with person's name